### PR TITLE
Migrate stack from nginx-proxy to traefik

### PR DESCRIPTION
**Pull request by Brutus5000**
- Enables dynamic settings for reverse proxy (e.g. adhoc blacklisting)
- Moves the exposing of services from the respective env to the
  service declaration in docker-compose.yml
- Adds a secured dashboard for the reverse proxy setup

### CODE REVIEW

File: config.template/faf-aio-replayserver/faf-aio-replayserver.env [hunk 1]
```diff
@@ -1,2 +1,17 @@
...
+REPLAY_TIMEOUT=300
+READ_BUFFER_SIZE=1024
+WRITE_BUFFER_SIZE=1024
+TEMPORARY_DIR=./tmp/
+REPLAY_DIR=/content/replays
+ENV=development
+APP_VERSION=0.4.11
+WAIT_STEP=5
+BUG_SNAG_API_KEY=
+
```
Comment (norraxx) [inline config.template/faf-aio-replayserver/faf-aio-replayserver.env]: I don't think, that you need that anymore.

Comment (norraxx) [inline config.template/faf-aio-replayserver/faf-aio-replayserver.env]: I don't think, that you need that anymore.

File: config.template/faf-aio-replayserver/faf-aio-replayserver.yml [hunk 2]
```diff
@@ -1,21 +0,0 @@
...
-    storage:
-        vault_path: /content/faf/vault/replay_vault
-    replay:
-        forced_end_time: "18000"
-        grace_period: "30"
-        delay:
-            replay_delay: "300"
-            update_interval: "1"
-        merge:
-            desired_quorum: "2"
```
Comment (norraxx) [inline config.template/faf-aio-replayserver/faf-aio-replayserver.yml]: Mazornoob won't like you :-D

Comment (norraxx) [reply config.template/faf-aio-replayserver/faf-aio-replayserver.yml]: I think, you'll need to do:
`git checkout develop`
`git pull`
`git checkout -`
`git rebase develop`

Comment (norraxx) [reply config.template/faf-aio-replayserver/faf-aio-replayserver.yml]: I think, you'll need to do:
`git checkout develop`
`git pull`
`git checkout -`
`git rebase develop`

File: docker-compose.yml [hunk 3]
```diff
@@ -205,7 +241,13 @@ services:
     networks:
       - faf
     ports:
-      - "127.0.0.1:8020:3000"
+      - "127.0.01:8020:3000" # For internal testing only
```
Comment (norraxx) [inline docker-compose.yml]: Missing dot.

File: docker-compose.yml [hunk 4]
```diff
@@ -179,11 +209,17 @@ services:
       - ./data/faf-java-api/logs:/logs
       - ./data/faf-java-api/repositories:/repositories
     expose:
+      - "8010" # HTTP API
       - "8011" # HTTP Management API
     ports:
-      - "8010:8010" # HTTP API
+      - "127.0.01:8010:8010" # For internal testing only
```
Comment (norraxx) [inline docker-compose.yml]: O_o copy-paste kung-fu with missing dots? or there is some magic in that IP addresses? O_o

File: docker-compose.yml [hunk 5]
```diff
@@ -166,7 +196,7 @@ services:
   #
   faf-java-api:
     container_name: faf-java-api
-    image: faforever/faf-java-api:v2.2.8
+    image: faforever/faf-java-api:v2.2.7
```
Comment (norraxx) [inline docker-compose.yml]: why?

**Review by norraxx — DISMISSED** [review]
rebase needed

File: docker-compose.yml (hunk 5 above)
Comment (Brutus5000) [reply docker-compose.yml]: Actually I did rebase and looks like that was the outcome xD. Will revert.

File: docker-compose.yml (hunk 4 above)
Comment (Brutus5000) [reply docker-compose.yml]: Good catch.

File: docker-compose.yml (hunk 3 above)
Comment (Brutus5000) [reply docker-compose.yml]: Good catch.

File: config.template/faf-aio-replayserver/faf-aio-replayserver.yml (hunk 2 above)
Comment (Brutus5000) [reply config.template/faf-aio-replayserver/faf-aio-replayserver.yml]: Yeah that rebase went horribly wrong.

File: docker-compose.yml [hunk 6]
```diff
@@ -15,6 +15,36 @@ services:
...
+      - "80:80"
+      - "443:443"
+      - "8080:8080" # The Web UI (enabled by --api)
+    networks:
+      - outside
+      - faf
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock # So that Traefik can listen to the Docker events
+      - ./config/traefik/traefik.toml:/etc/traefik/traefik.toml # Only general config, service specific stuff via label
+      - ./config/traefik/traefik.toml:/etc/traefik/rules.toml # Change dynamic rules here
```
Comment (norraxx) [inline docker-compose.yml]: traefik.toml -> rules.toml ?

File: docker-compose.yml [hunk 7]
```diff
@@ -593,7 +609,7 @@ services:
   #
   faf-unitdb:
     container_name: faf-unitdb
-    image: faforever/faf-unitdb:v1.3
+    image: faforever/faf-unitdb:v1.2.11
```
Comment (norraxx) [inline docker-compose.yml]: downgrade? Rackover won't be happy

File: docker-compose.yml [hunk 8]
```diff
@@ -229,6 +271,11 @@ services:
       interval: 30s
       timeout: 5s
       retries: 3
+    labels:
+      - "traefik.enable=true"
+      - "traefik.backend=spring-boot-admin"
+      - "traefik.port=9090"
```
Comment (norraxx) [inline docker-compose.yml]: I've already seen 9090 at prometheus. 1111?

File: docker-compose.yml [hunk 9]
```diff
@@ -473,22 +539,19 @@ services:
   #
   faf-aio-replayserver:
     container_name: faf-aio-replayserver
-    image: faforever/faf-aio-replayserver:0.8.3
+    # build: git://github.com/FAForever/faf-aio-replayserver#0.1
+    image: faforever/faf-aio-replayserver:0.4.11
```
Comment (norraxx) [inline docker-compose.yml]: I'm still thinking, that you're reverting mazornoob changes @ replay server

File: docker-compose.yml (hunk 7 above)
Comment (Brutus5000) [reply docker-compose.yml]: fixed

File: docker-compose.yml (hunk 9 above)
Comment (Brutus5000) [reply docker-compose.yml]: fixed

**Review by Sheeo — APPROVED** [review]

**Pull request merged**